src/test/scala/com/fulcrumgenomics/bam/SamRecordClipperTest.scala

```diff
@@ -588,24 +588,74 @@ class SamRecordClipperTest extends UnitSpec with OptionValues {
     mate.cigar.toString shouldBe "1S99M"
   }
 
+  it should "clip reads that overlap with soft-clipping on the forward read after the midpoint" in {
+    val (rec, mate) = pair(1, "95M5S", Plus, 50, "100M", Minus)
+    clipper(Hard).clipOverlappingReads(rec=rec, mate=mate) shouldBe (23, 23)
+    rec.start shouldBe 1
+    rec.end shouldBe 72
+    rec.cigar.toString shouldBe "72M28H"
+    mate.start shouldBe 73
+    mate.cigar.toString shouldBe "23H77M"
+  }
+
+  it should "clip reads that overlap with soft-clipping on the reverse read before the midpoint" in {
+    val (rec, mate) = pair(1, "100M", Plus, 55, "5S95M", Minus)
+    clipper(Hard).clipOverlappingReads(rec=rec, mate=mate) shouldBe (23, 23)
+    rec.start shouldBe 1
+    rec.end shouldBe 77
+    rec.cigar.toString shouldBe "77M23H"
+    mate.start shouldBe 78
+    mate.cigar.toString shouldBe "28H72M"
+  }
+
+  it should "clip reads that overlap 1 bp directly in the middle of the pair" in {
+    val (rec, mate) = pair(1, "99M1S", Plus, 99, "1S99M", Minus)
+    clipper(Hard).clipOverlappingReads(rec=rec, mate=mate) shouldBe (0, 1)
+    rec.start shouldBe 1
+    rec.end shouldBe 99
+    rec.cigar.toString shouldBe "99M1H"
+    mate.start shouldBe 100
+    mate.cigar.toString shouldBe "2H98M"
+  }
+
+  it should "clip reads that overlap in the first half of the pair, not over the middle" in {
+    val (rec, mate) = pair(1, "95M5S", Plus, 90, "20S80M", Minus)
+    clipper(Hard).clipOverlappingReads(rec=rec, mate=mate) shouldBe (3, 3)
+    rec.start shouldBe 1
+    rec.end shouldBe 92
+    rec.cigar.toString shouldBe "92M8H"
+    mate.start shouldBe 93
+    mate.cigar.toString shouldBe "23H77M"
+  }
+
+  it should "clip reads that overlap in the second half of the pair, not over the middle" in {
+    val (rec, mate) = pair(1, "80M20S", Plus, 70, "5S95M", Minus)
+    clipper(Hard).clipOverlappingReads(rec=rec, mate=mate) shouldBe (5, 6)
+    rec.start shouldBe 1
+    rec.end shouldBe 75
+    rec.cigar.toString shouldBe "75M25H"
+    mate.start shouldBe 76
+    mate.cigar.toString shouldBe "11H89M"
+  }
+
   it should "clip reads that overlap with one end having a deletion" in {
     val (rec, mate) = pair(1, "60M10D40M", Plus, 50, "10M10D80M10D10M", Minus)
-    clipper(Soft).clipOverlappingReads(rec=rec, mate=mate) shouldBe (25, 26)
+    clipper(Soft).clipOverlappingReads(rec=rec, mate=mate) shouldBe (30, 21)
```

Comment:
I think you're alternative midpoint calculation will revert the # of bases clipped off of each end, so the two values are as close as possible.